lib/include/openamp/remoteproc_virtio.h

```diff
@@ -23,6 +24,19 @@ extern "C" {
 /* maximum number of vring descriptors for a vdev limited by 16-bit data type */
 #define	RPROC_MAX_VRING_DESC	USHRT_MAX
 
+/* cache invalidation helpers for resource table */
+#ifdef VIRTIO_CACHED_RSC_TABLE
+#define RSC_TABLE_FLUSH(x) \
+	({void *_x = (x); \
+	metal_cache_flush(&_x, sizeof(_x)); })
```

Comment:
Are you sure this is doing what you think it is doing? Looking at it it looks like you are flushing the pointer here, not the actual struct also with a wrong `sizeof()`.

Reply:
You're right. I've fixed it.
Also, to pass checkpatch I've added 2 arguments for RSC_TABLE_FLUSH and RSC_TABLE_INVALIDATE